connection.go

```diff
@@ -89,10 +89,30 @@ func processSetStatement(ctx context.Context, c *fireboltConnection, query strin
 		return false, nil
 	}
 
-	_, err = c.client.Query(ctx, c.engineUrl, c.databaseName, "SELECT 1", map[string]string{setKey: setValue})
+	if setKey == "database" {
+		return true, fmt.Errorf("`SET database` query is not allowed. Please use `USE database` syntax")
+	}
+
+	parameters := map[string]string{setKey: setValue}
+	if db, ok := c.parameters["database"]; ok {
+		parameters["database"] = db
+	}
+	_, err = c.client.Query(ctx, c.engineUrl, "SELECT 1", parameters, c.setParameter)
 	if err == nil {
-		c.setStatements[setKey] = setValue
+		c.setParameter(setKey, setValue)
 		return true, nil
 	}
 	return true, err
 }
+
+func (c *fireboltConnection) setParameter(key, value string) {
+	if c.parameters == nil {
+		c.parameters = make(map[string]string)
+	}
+	c.parameters[key] = value
+	// Cache parameter in connector as well in case connection will be recreated by the pool
+	if c.connector.cachedParameters == nil {
+		c.connector.cachedParameters = make(map[string]string)
+	}
+	c.connector.cachedParameters[key] = value
```

Comment:
Not familiar with this process in go, does this mean the same connection will be recreated by the connection pool? Or do you mean to cache those parameters for all consecutive connections? If the latter I'm not sure if it's expected - when I open a connection I would expect a "blank slate" with no cached parameters.

Reply:
We cache parameters in a `FireboltConnector` object
This object is something in the middle between a driver and a connection
The problem is that a connection can be recreated from scratch when `Exec` is called, since by default DSN string is considered a full definition of a connection (two connections are identical if their DSNs are the same). Which is not the case for us since we have parameters that are accumulated. For this purpose, a `FireboltConnector` was added. It overrides the logic of creating a new connection for a DSN, and takes care of preserving accumulated parameters. This object is created once per `sql.Open` call, so parameters are isolated between a logical connection

Reply:
Makes sense. Would then `FireboltSession` be a better name for `FireboltConnector`? Or do you think it will be confusing?